meilisearch-auth/src/store.rs

```diff
@@ -92,13 +92,68 @@ impl HeedAuthStore {
             Action::into_enum_iter().collect()
         } else if key.actions.contains(&Action::DocumentsAll) {
             // if key.actions.contains.DocumentsAll add all actions related to documents.
-            let mut actions = key.actions.clone();
-            actions.append(&mut vec![
-                Action::DocumentsAdd,
-                Action::DocumentsGet,
-                Action::DocumentsDelete,
-            ]);
-            actions
+            key.actions
+                .iter()
+                .cloned()
+                .filter(|action| {
+                    // Prevents duplicate entries in the actions vector
+                    *action != Action::DocumentsAdd
+                        && *action != Action::DocumentsDelete
+                        && *action != Action::DocumentsGet
+                })
+                .chain(vec![
+                    Action::DocumentsAdd,
+                    Action::DocumentsDelete,
+                    Action::DocumentsGet,
+                ])
+                .collect()
+        } else if key.actions.contains(&Action::IndexesAll) {
+            key.actions
+                .iter()
+                .cloned()
+                .filter(|action| {
+                    *action != Action::IndexesAdd
+                        && *action != Action::IndexesGet
+                        && *action != Action::IndexesDelete
+                        && *action != Action::IndexesUpdate
+                })
+                .chain(vec![
+                    Action::IndexesAdd,
+                    Action::IndexesGet,
+                    Action::IndexesDelete,
+                    Action::IndexesUpdate,
+                ])
+                .collect()
```

Comment:
It was okay-ish to do a contain when there was only one, but now we’re iterating over and over on all the actions, and it’s taking a lot of lines of code.

I thought it would be easier to iterate on all the actions, transform the `.*` into what we want, and let the other as-is.

And finally, to remove the duplicate entry, I would prefer a simpler collect to a HashSet + a conversion to a vec at the very end of the computation.

Something like that;
```rust
let actions = key.actions.iter().flat_map(|action| match action {
    IndexesAll => [IndexesAdd, IndexesGet, IndexesDelete, IndexesUpdate].iter(),
    SettingsAll => todo!(),
    *All => todo!(),
...
    other => std::iter::once(other),
  })
  .collect::<HashSet<Action>>();

let action = actions.into_iter().collect::<Vec<Action>>();
```

(this code probably has issues with the iterator in the match, but you get the idea)